{
	"ID": "ALAS-2019-1324",
	"Author": "linux-security@amazon.com",
	"From": "linux-security@amazon.com",
	"Status": "final",
	"Type": "security",
	"Version": "1.4",
	"Title": "Amazon Linux AMI 2014.03 - ALAS-2019-1324: important priority package update for python34",
	"Issued": {
		"Date": "2019-11-19 17:31"
	},
	"Updated": {
		"Date": "2019-11-22 03:19"
	},
	"Severity": "important",
	"Description": "Package updates are available for Amazon Linux AMI that fix the following vulnerabilities:\nCVE-2019-9948:\n\turllib in Python 2.x through 2.7.16 supports the local_file: scheme, which makes it easier for remote attackers to bypass protection mechanisms that blacklist file: URIs, as demonstrated by triggering a urllib.urlopen(&#039;local_file:///etc/passwd&#039;) call.\n1695570: \nCVE-2019-9948 python: Undocumented local_file protocol allows remote attackers to bypass protection mechanisms\n\nCVE-2019-9947:\n\tAn issue was discovered in urllib2 in Python 2.x through 2.7.16 and urllib in Python 3.x through 3.7.3. CRLF injection is possible if the attacker controls a url parameter, as demonstrated by the first argument to urllib.request.urlopen with \\r\\n (specifically in the path component of a URL that lacks a ? character) followed by an HTTP header or a Redis command. This is similar to the CVE-2019-9740 query string issue.\n1695572: \nCVE-2019-9947 python: improper neutralization of CRLF sequences in urllib module\n1695572: \nCVE-2019-9947 python: CRLF injection via the path part of the url passed to urlopen()\n\nCVE-2019-9740:\n\tAn issue was discovered in urllib2 in Python 2.x through 2.7.16 and urllib in Python 3.x through 3.7.3. CRLF injection is possible if the attacker controls a url parameter, as demonstrated by the first argument to urllib.request.urlopen with \\r\\n (specifically in the query string after a ? character) followed by an HTTP header or a Redis command.\n1688169: \nCVE-2019-9740 python: improper neutralization of CRLF sequences in urllib module\n1688169: \nCVE-2019-9740 python: CRLF injection via the query part of the url passed to urlopen()\n\nCVE-2019-9636:\n\tPython 2.7.x through 2.7.16 and 3.x through 3.7.2 is affected by: Improper Handling of Unicode Encoding (with an incorrect netloc) during NFKC normalization. The impact is: Information disclosure (credentials, cookies, etc. that are cached against a given hostname). The components are: urllib.parse.urlsplit, urllib.parse.urlparse. The attack vector is: A specially crafted URL could be incorrectly parsed to locate cookies or authentication data and send that information to a different host than when parsed correctly.\n1688543: \nCVE-2019-9636 python: Information Disclosure due to urlsplit improper NFKC normalization\n\nCVE-2019-16056:\n\tAn issue was discovered in Python through 2.7.16, 3.x through 3.5.7, 3.6.x through 3.6.9, and 3.7.x through 3.7.4. The email module wrongly parses email addresses that contain multiple @ characters. An application that uses the email module and implements some kind of checks on the From/To headers of a message could be tricked into accepting an email address that should be denied. An attack may be the same as in CVE-2019-11340; however, this CVE applies to Python more generally.\n99999: \nCVE-2019-16056 python: email.utils.parseaddr wrongly parses email addresses\n\nCVE-2019-10160:\n\tA security regression of CVE-2019-9636 was discovered in python, since commit d537ab0ff9767ef024f26246899728f0116b1ec3, which still allows an attacker to exploit CVE-2019-9636 by abusing the user and password parts of a URL. When an application parses user-supplied URLs to store cookies, authentication credentials, or other kind of information, it is possible for an attacker to provide specially crafted URLs to make the application locate host-related information (e.g. cookies, authentication data) and send them to a different host than where it should, unlike if the URLs had been correctly parsed. The result of an attack may vary based on the application.\n1718388: \nCVE-2019-10160 python: regression of CVE-2019-9636 due to functional fix to allow port numbers in netloc\n",
	"References": {
		"Reference": [
			{
				"Href": "http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-10160",
				"ID": "CVE-2019-10160",
				"Title": "",
				"Type": "cve"
			},
			{
				"Href": "http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-16056",
				"ID": "CVE-2019-16056",
				"Title": "",
				"Type": "cve"
			},
			{
				"Href": "http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-9636",
				"ID": "CVE-2019-9636",
				"Title": "",
				"Type": "cve"
			},
			{
				"Href": "http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-9740",
				"ID": "CVE-2019-9740",
				"Title": "",
				"Type": "cve"
			},
			{
				"Href": "http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-9947",
				"ID": "CVE-2019-9947",
				"Title": "",
				"Type": "cve"
			},
			{
				"Href": "http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-9948",
				"ID": "CVE-2019-9948",
				"Title": "",
				"Type": "cve"
			}
		]
	},
	"Pkglist": {
		"Collection": {
			"Short": "amazon-linux-ami",
			"Name": "Amazon Linux AMI",
			"Package": [
				{
					"Arch": "x86_64",
					"Epoch": "0",
					"Name": "python34-debuginfo",
					"Release": "1.49.amzn1",
					"Version": "3.4.10",
					"Filename": "Packages/python34-debuginfo-3.4.10-1.49.amzn1.x86_64.rpm"
				},
				{
					"Arch": "x86_64",
					"Epoch": "0",
					"Name": "python34-test",
					"Release": "1.49.amzn1",
					"Version": "3.4.10",
					"Filename": "Packages/python34-test-3.4.10-1.49.amzn1.x86_64.rpm"
				},
				{
					"Arch": "x86_64",
					"Epoch": "0",
					"Name": "python34-devel",
					"Release": "1.49.amzn1",
					"Version": "3.4.10",
					"Filename": "Packages/python34-devel-3.4.10-1.49.amzn1.x86_64.rpm"
				},
				{
					"Arch": "x86_64",
					"Epoch": "0",
					"Name": "python34-libs",
					"Release": "1.49.amzn1",
					"Version": "3.4.10",
					"Filename": "Packages/python34-libs-3.4.10-1.49.amzn1.x86_64.rpm"
				},
				{
					"Arch": "x86_64",
					"Epoch": "0",
					"Name": "python34-tools",
					"Release": "1.49.amzn1",
					"Version": "3.4.10",
					"Filename": "Packages/python34-tools-3.4.10-1.49.amzn1.x86_64.rpm"
				},
				{
					"Arch": "x86_64",
					"Epoch": "0",
					"Name": "python34",
					"Release": "1.49.amzn1",
					"Version": "3.4.10",
					"Filename": "Packages/python34-3.4.10-1.49.amzn1.x86_64.rpm"
				},
				{
					"Arch": "i686",
					"Epoch": "0",
					"Name": "python34-devel",
					"Release": "1.49.amzn1",
					"Version": "3.4.10",
					"Filename": "Packages/python34-devel-3.4.10-1.49.amzn1.i686.rpm"
				},
				{
					"Arch": "i686",
					"Epoch": "0",
					"Name": "python34-test",
					"Release": "1.49.amzn1",
					"Version": "3.4.10",
					"Filename": "Packages/python34-test-3.4.10-1.49.amzn1.i686.rpm"
				},
				{
					"Arch": "i686",
					"Epoch": "0",
					"Name": "python34",
					"Release": "1.49.amzn1",
					"Version": "3.4.10",
					"Filename": "Packages/python34-3.4.10-1.49.amzn1.i686.rpm"
				},
				{
					"Arch": "i686",
					"Epoch": "0",
					"Name": "python34-debuginfo",
					"Release": "1.49.amzn1",
					"Version": "3.4.10",
					"Filename": "Packages/python34-debuginfo-3.4.10-1.49.amzn1.i686.rpm"
				},
				{
					"Arch": "i686",
					"Epoch": "0",
					"Name": "python34-libs",
					"Release": "1.49.amzn1",
					"Version": "3.4.10",
					"Filename": "Packages/python34-libs-3.4.10-1.49.amzn1.i686.rpm"
				},
				{
					"Arch": "i686",
					"Epoch": "0",
					"Name": "python34-tools",
					"Release": "1.49.amzn1",
					"Version": "3.4.10",
					"Filename": "Packages/python34-tools-3.4.10-1.49.amzn1.i686.rpm"
				}
			]
		}
	}
}
